hamocc/mo_ini_fields.F90

```diff
@@ -5,324 +5,302 @@
+    !******************************************************************************
 
-      implicit none
+    use mo_carbch,      only: co2star,co3,hi,ocetra
+    use mo_param_bgc,   only: fesoly,cellmass,fractdim,bifr13,bifr14,c14fac,re1312,re14to
+    use mo_biomod,      only: abs_oce
+    use mo_control_bgc, only: rmasks,use_FB_BGC_OCE, use_cisonew, use_AGG, use_CFC, use_natDIC, use_BROMO, use_sedbypass
+    use mo_param1_bgc,  only: ialkali,ian2o,iano3,icalc,idet,idicsat,idms,idoc,ifdust,igasnit,iiron,iopal,ioxygen,iphosph,iphy, &
+                              iprefalk,iprefdic,iprefo2,iprefpo4,isco212,isilica,izoo, &
+                              iadust,inos,ibromo,icfc11,icfc12,isf6, &
+                              icalc13,icalc14,idet13,idet14,idoc13,idoc14,iphy13,iphy14,isco213,isco214,izoo13,izoo14,safediv, &
+                              inatcalc, &
+                              ipowaal,ipowaic,ipowaox,ipowaph,ipowasi,ipown2,ipowno3,isssc12,issso12,issssil,issster,ks,nsedtra, &
+                              ipowc13,ipowc13,issso13,issso13,isssc13,ipowc14,isssc14,issso14
+    use mo_vgrid,       only: kmle,kbo
+    use mo_carbch,      only: nathi,natco3
+    use mo_sedmnt,      only: sedhpl,burial,powtra,sedlay
+    use mo_profile_gd,  only: profile_gd
 
-      INTEGER, intent(in) :: kpaufr,kpie,kpje,kpke,kbnd
-      REAL,    intent(in) :: pddpo(kpie,kpje,kpke)
-      REAL,    intent(in) :: prho (kpie,kpje,kpke)
-      REAL,    intent(in) :: omask(kpie,kpje)
-      REAL,    intent(in) :: pglon(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)
-      REAL,    intent(in) :: pglat(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)
+    ! Arguments
+    integer, intent(in) :: kpaufr                                   ! 1/0 flag, 1 indicating a restart run
+    integer, intent(in) :: kpie                                     ! 1st dimension of model grid.
+    integer, intent(in) :: kpje                                     ! 2nd dimension of model grid.
+    integer, intent(in) :: kpke                                     ! 3rd (vertical) dimension of model grid.
+    integer, intent(in) :: kbnd                                     ! nb of halo grid points
+    real,    intent(in) :: pddpo(kpie,kpje,kpke)                    ! size of grid cell (3rd dimension) [m].
+    real,    intent(in) :: prho (kpie,kpje,kpke)                    ! density [g/cm^3].
+    real,    intent(in) :: omask(kpie,kpje)                         ! ocean mask.
+    real,    intent(in) :: pglon(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd) ! longitude of grid cell [deg].
+    real,    intent(in) :: pglat(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd) ! latitude  of grid cell [deg].
 
-      ! local variables
-      INTEGER :: i,j,k,l
-      REAL :: rco213,rco214,beta13,beta14 ! cisonew
-      REAL :: snow ! AGG
+    ! local variables
+    integer :: i,j,k,l
+    real    :: rco213,rco214,beta13,beta14 ! cisonew
+    real    :: snow ! agg
```

Comment:
`agg` could be capitalized again, since it refers to the logical `AGG`

Reply:
Done